### DISCUSSION TITLE
r".." not support in map initialization

### DISCUSSION THREAD

**Discussion by jdonnerstag** [vlang/v#14619] [Ideas]
I can do:
``` 
data := {
    '"\\x00"': [byte(0)]
    '"\\x01"': [byte(0x01)]
    '"\\u2603"': [byte(0xe2), 0x98, 0x83]
    '"\\xE2\\x98\\x83"': [byte(0xe2), 0x98, 0x83]
}
```
but not 
```
data := {
    r'"\x00"': [byte(0)]
    r'"\x01"': [byte(0x01)]
    r'"\u2603"': [byte(0xe2), 0x98, 0x83]
    r'"\xE2\x98\x83"': [byte(0xe2), 0x98, 0x83]
}
```
That is, the `r".."` approach to create raw strings is not supported in map initialisation. This leads to a compiler error.

**Comment by JalonSolov** [comment]
Even if it accepted the `r` strings, you wouldn't get what you expect.  "raw" strings are just that - raw.  No translation, escaping, etc. is done, so you would wind up with the first element of `data` as a literal string of `"`, `\`', `x`, `0`, `0`, `"`.

**Comment by JalonSolov** [comment]
The error without `r` strings is
```
foo.v:2:9: error: cannot use `\x00` (NULL character) in the string literal
    1 | data := {
    2 |     '\x00': [byte(0)]
      |         ^
    3 |     '\x01': [byte(0x01)]
    4 |     '\u2603': [byte(0xe2), 0x98, 0x83]
```
the error _with_ `r` strings is
```
foo.v:2:6: error: unexpected string `\x00`, expecting `:`
    1 | data := {
    2 |     r'\x00': [byte(0)]
      |      ~~~~~~
    3 |     r'\x01': [byte(0x01)]
    4 |     r'\u2603': [byte(0xe2), 0x98, 0x83]
```

**Comment by jdonnerstag** [comment]
It is *exactly* what I want: A string literal, which is "raw" and which the vlang parser did not try to unescape. 

You may also argue that V currently is not consistent. In some place raw string are allowed, and in others it not. With no reason why, and no mention in the docs.

**Comment by JalonSolov** [comment]
I agree that it should be supported here, and if that is what you want, then you will get what you want... once it is supported.
